docs/user/using-an-OIDC-provider.md

```diff
@@ -22,444 +22,29 @@ When configuring the identity provider, you will need to ensure that the redirec
-
-Once you can reproduce the issue, there are a couple of possibilities for the cause which commonly trip people up.
-One common issue is that the Kubernetes cluster's api server is not configured for oidc (some people don't realise this is necessary). This is easy to check by grepping for `oidc` in the api server pod output, for example, if your cluster _is_ configured for OpenID Connect, you should see something like:
-
-```bash
-$ kubectl -n kube-system get po -l component=kube-apiserver -o yaml | grep oidc
-      - --oidc-ca-file=/etc/kubernetes/pki/apiserver.crt
-      - --oidc-client-id=default
-      - --oidc-groups-claim=groups
-      - '--oidc-groups-prefix=oidc:'
-      - --oidc-issuer-url=https://172.18.0.2:32000
-      - --oidc-username-claim=email
-      - '--oidc-username-prefix=oidc:'
-```
-
-### Checking your OIDC prefix
-
-Another common point of confusion is the `--oidc-username-prefix` option specified above. If it is set to a value such as `oidc:` (or using a [default value](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#configuring-the-api-server) without being set explicitly), then a JWT token identifying a user of `myname@example.com` will require the RBAC RoleBinding to specify the user as `oidc:myname@example.com` to match the OIDC user with the defined RBAC.
-
-### Checking the logs of your Kubernetes API server
-
-Finally, if none of the above are relevant to your issue, you can check the logs of the Kubernetes API server deployment for OIDC-related lines at the time of your login attempt. These may show a configuration issue with the API server itself.
-
-### User automatically logged out from Kubeapps Console
+Kubeapps uses [OAuth2 Proxy](https://github.com/oauth2-proxy/oauth2-proxy) to handle the OAuth2/OpenIDConnect authentication. The following sections explain how you can find the parameters above for some of the identity providers tested. If you have configured your cluster to use an Identity Provider you will already know some of these parameters. More detailed information can be found on the [OAuth2 Proxy Auth configuration page](https://oauth2-proxy.github.io/oauth2-proxy/auth-configuration).
 
-When using the default auth proxy, some users may experience the behavior where they are automatically logged out from the console.
-Prior to the Kubeapps chart version 7.1.0, the auth proxy configuration did not include a default `--cookie-refresh` value to refresh the access/openid token and so the console will logout once the token expires. In the case of Keycloak for example, this can happen quickly as the default access token expiration is 5m.
+- [VMware Cloud Services](./OAuth2OIDC-VMware-cloud-services.md)
+- [Azure Active Directory](./OAuth2OIDC-azure-active-directory.md)
+- [Google OpenID Connect](./OAuth2OIDC-google-openid-connect.md)
+- [Keycloak](./OAuth2OIDC-keycloak.md)
+- [Dex](OAuth2OIDC-dex.md)
+- [Oauth2-proxy](./OAuth2OIDC-oauth2-proxy.md)
```

Comment:
oauth2-proxy doesn't really belong like this as another identity provider - it's not one at all. We use oauth2-proxy to proxy oauth2 calls to an identity provider. I'd probably leave the oauth2-proxy details in the main file still?